src/filters/pages.js

```diff
@@ -193,13 +193,13 @@ params:
     type: string
 */
 const render = (page, key) => {
+  /* will pages ever not have data attributes? */
```

Comment:
Ah, found it. https://www.11ty.dev/docs/collections/#collection-item-data-structure

So I guess pages _within a collection_ have stuff on `page.data`, but pages outside of a collection have their stuff on just `page`?

I have some concerns about the maintainability of the current structure, with a mix of Nunjucks and Markdown, lots and lots of JS filters and lots of Nunjucks macros. There are so many small pieces (which can be a plus) -- in practice it's very difficult for me to understand what any one piece is actually doing. It feels like lots of rabbit holes.

I'll try to think a bit more about what precisely is the issue I'm trying to articulate, and what I'd recommend to resolve it.

Reply:
Yeah, I find the difference between the two totally frustrating. I agree there's a lot going on. I'm not sure I know the fix…

Reply:
I will say it's not always clear to me what the data looks like (for things like pages and collections) and that sort of makes me feel the testing is brittle (if that makes sense). I may be worse at articulating what exactly my frustration is, so  I am interesting to hear you all's thoughts on how things are structured.

Reply:
There are two page objects, the link Jonny shred (a collection-page) and the current [`page` variable](https://www.11ty.dev/docs/data-eleventy-supplied/#page-variable-contents). They have a lot of overlap (`url`, `date`, `inputPath`, `outputPath`, `fileSlug`), but a few differences:

- collection-pages have `templateContent` and `data` (which includes page, directory, and global data)
- current `page` has `filePathStem`, but the content and data-properties are not included. We have to access them un-scoped as `content`, and `<property>`.

I find it very tempting to do `set page = collections.all | getPage(page.url)` first thing in our layouts, so that the page object includes all data and content. We would just have to make sure that layout is not used by any page excluded from collections. (or maybe there's no reason to exclude pages from collections…)